hal/src/common/thumbv6m/sercom/v2/spi.rs

```diff
@@ -0,0 +1,1768 @@
+            pads,
+            mode: PhantomData,
+            chsize: PhantomData,
+            freq: freq.into(),
+        }
+    }
+
+    /// Create a new [`Config`] in the default configuration
+    ///
+    /// This function will enable the corresponding APB clock, reset the
+    /// [`Sercom`] peripheral, and return a [`Config`] in the default
+    /// configuration. The [`Config`] takes ownership of the [`Sercom`] and
+    /// [`Pads`].
+    ///
+    /// Users must configure GCLK manually. The `freq` parameter represents the
+    /// GCLK frequency for this [`Sercom`] instance.
+    #[inline]
+    pub fn new(pm: &PM, mut sercom: P::Sercom, pads: P, freq: impl Into<Hertz>) -> Self {
+        sercom.enable_apb_clock(pm);
+        Self::create(sercom, pads, freq)
+    }
+}
+
+impl<P, M, C> Config<P, M, C>
+where
+    P: DipoDopo,
+    M: Mode,
+    C: CharSize,
+{
+    /// Reset the SERCOM peripheral
+    #[inline]
+    fn swrst(sercom: &P::Sercom) {
+        sercom.spi().ctrla.write(|w| w.swrst().set_bit());
+        while sercom.spi().syncbusy.read().swrst().bit_is_set() {}
+        unsafe { Self::reset_serial_read_state() };
+    }
+
+    /// Reset the `static` state used by the implementation of [`Read`]
+    ///
+    /// The state should be reset whenever the peripheral is reset or disabled.
+    /// It should not be possible for the state to become out of sync from safe
+    /// code.
+    unsafe fn reset_serial_read_state() {
+        SERIAL_READ_STATE[P::Sercom::NUM] = false;
+    }
+
+    /// Change the [`Config`] [`Mode`] or [`CharSize`]
+    #[inline]
+    fn change<M2, C2>(self) -> Config<P, M2, C2>
+    where
+        M2: Mode,
+        C2: CharSize,
+    {
+        Config {
+            sercom: self.sercom,
+            pads: self.pads,
+            mode: PhantomData,
+            chsize: PhantomData,
+            freq: self.freq,
+        }
+    }
+
+    /// Trigger the [`Sercom`]'s SWRST and return a [`Config`] in the
+    /// default configuration.
+    #[inline]
+    pub fn reset(self) -> Config<P> {
+        Config::create(self.sercom, self.pads, self.freq)
+    }
+
+    /// Obtain a reference to the PAC `SERCOM` struct
+    ///
+    /// Directly accessing the `SERCOM` could break the invariants of the
+    /// type-level tracking in this module, so it is unsafe.
+    #[inline]
+    pub unsafe fn sercom(&self) -> &P::Sercom {
+        &self.sercom
+    }
+
+    /// Consume the [`Config`], reset the peripheral, and return the [`Sercom`]
+    /// and [`Pads`]
+    #[inline]
+    pub fn free(self) -> (P::Sercom, P) {
+        Self::swrst(&self.sercom);
+        (self.sercom, self.pads)
+    }
+
+    /// Change the operating [`Mode`]
+    #[inline]
+    pub fn op_mode<M2: Mode>(self) -> Config<P, M2, C> {
+        M2::configure(&self.sercom);
+        self.change()
+    }
+
+    /// Change the [`CharSize`]
+    #[inline]
+    pub fn char_size<C2: CharSize>(self) -> Config<P, M, C2> {
+        C2::configure(&self.sercom);
```

Comment:
It might be useful to specify the default here?

Reply:
Good catch

Reply:
Wait, what do you mean by default? Do you mean the default `Config` returned by `new`? Looks like I forgot to document the default `CharSize` there. But there's no default for the `char_size` function. You have to provide the desired type.

Reply:
Right now, the default `CharSize` is documented in the module-level docs and the docs for `Config`, but it is not documented in the `Config::new` function. I added it there as well. Is that sufficient?

Reply:
SGTM